appstudio-controller/controllers/appstudio.redhat.com/sandboxprovisioner_controller.go

```diff
@@ -190,20 +190,15 @@ func createSpaceRequestForDTC(ctx context.Context, k8sClient client.Client, dtc
 			TargetClusterRoles: []string{}, // To be updated in the future once cluster roles become defined
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: dtc.Name,
+			GenerateName: dtc.Name + "-",
 			Namespace:    dtc.Namespace,
 			Labels: map[string]string{
 				deploymentTargetClaimLabel: dtc.Name,
 			},
 		},
 	}
 
-	err := ctrl.SetControllerReference(dtc, &newSpaceRequest, k8sClient.Scheme())
-	if err != nil {
-		return nil, err
-	}
-
-	err = k8sClient.Create(ctx, &newSpaceRequest)
+	err := k8sClient.Create(ctx, &newSpaceRequest)
```

Comment:
As above, removed the owner ref as it was interfering with expected behaviour in some cases.